### ISSUE TITLE
[ES6] Parse error with quotes for object key in shorthand function syntax.

### ISSUE THREAD

**Issue by laziel** [google/closure-compiler#2705]
A shorter syntax for method definitions on objects (ES6) is not parsed expectedly 
if any type of "quote" exists for an object key.

```
java -jar target/closure-compiler-1.0-SNAPSHOT.jar 
--language_in=ECMASCRIPT_NEXT
--compilation_level=ADVANCED_OPTIMIZATIONS
--js test.js
```
[test.js:](https://closure-compiler-debugger.appspot.com/#input0%3Dvar%2520obj%2520%253D%2520%257B%250A%2520%2520%2520%2520%2522run%2522()%2520%257B%250A%2520%2520%2520%2520%2520%2520%2520%2520%252F%252F%2520this%2520is%2520a%2520function%250A%2520%2520%2520%2520%257D%250A%257D%253B%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3Dtrue%26CHECK_SYMBOLS%3Dtrue%26MISSING_PROPERTIES%3Dtrue%26TRANSPILE%3Dtrue%26CHECK_TYPES%3Dtrue%26CLOSURE_PASS%3Dtrue%26PRESERVE_TYPE_ANNOTATIONS%3Dtrue%26PRETTY_PRINT%3Dtrue) // input0:2: ERROR - Parse error. '[' expected
```js
var obj = {
    "run"() {
        // this is a function
    }
};
```

If the object key defined [without quotes](https://closure-compiler-debugger.appspot.com/#input0%3Dvar%2520obj%2520%253D%2520%257B%250A%2520%2520%2520%2520run()%2520%257B%250A%2520%2520%2520%2520%2520%2520%2520%2520%252F%252F%2520this%2520is%2520a%2520function%250A%2520%2520%2520%2520%257D%250A%257D%253B%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3Dtrue%26CHECK_SYMBOLS%3Dtrue%26MISSING_PROPERTIES%3Dtrue%26TRANSPILE%3Dtrue%26CHECK_TYPES%3Dtrue%26CLOSURE_PASS%3Dtrue%26PRESERVE_TYPE_ANNOTATIONS%3Dtrue%26PRETTY_PRINT%3Dtrue) or the object value is defined in [arrow function](https://closure-compiler-debugger.appspot.com/#input0%3Dvar%2520obj%2520%253D%2520%257B%250A%2520%2520%2520%2520%2522run%2522%253A%2520()%2520%253D%253E%2520%257B%250A%2520%2520%2520%2520%2520%2520%2520%2520%252F%252F%2520this%2520is%2520a%2520function%250A%2520%2520%2520%2520%257D%250A%257D%253B%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3Dtrue%26CHECK_SYMBOLS%3Dtrue%26MISSING_PROPERTIES%3Dtrue%26TRANSPILE%3Dtrue%26CHECK_TYPES%3Dtrue%26CLOSURE_PASS%3Dtrue%26PRESERVE_TYPE_ANNOTATIONS%3Dtrue%26PRETTY_PRINT%3Dtrue), it works:
```js
var obj = {
    "run": () => {
        // this is a function
    }
};
```

**Comment by lauraharker** [comment]
Looks like this is a known bug - see https://github.com/google/closure-compiler/issues/2566.

(Also, you may already know this, but I'll note that arrow functions can be dangerous to use as methods, since they have a lexically scoped `this`.)

**Comment by brad4d** [comment]
closed as a duplicate of #2566

**Comment by laziel** [comment]
@lauraharker @brad4d Thank you all! and sorry for my poor search :cry:
